src/xenia/kernel/xbdm/xbdm_misc.cc

```diff
@@ -171,6 +171,66 @@ dword_result_t DmGetSystemInfo_entry(pointer_t<XBDM_SYSTEM_INFO> info) {
 }
 DECLARE_XBDM_EXPORT1(DmGetSystemInfo, kDebug, kStub);
 
+dword_result_t DmSetMemory_entry(lpvoid_t dest, dword_t buf_size, lpvoid_t src,
+                                 lpdword_t bytes_written) {
+  if (!dest || !src || !buf_size) {
+    return XBDM_UNSUCCESSFUL;
+  }
+
+  size_t length = memory::page_size();
+  memory::PageAccess access = memory::PageAccess::kNoAccess;
+  BaseHeap* heap = kernel_state()->memory()->LookupHeap(dest.guest_address());
+  memory::QueryProtect(heap, length, access);
```

Comment:
heap->QueryProtect or even ``QueryRangeAccess`` as it allows you to just provide low and high address and it returns PageAccess